src/solver/optimisation/adequacy_patch_csr/set_problem_cost_function.cpp

```diff
@@ -113,47 +116,50 @@ void setLinearCost(PROBLEME_HEBDO* ProblemeHebdo, int hour)
         TransportCost = ProblemeHebdo->CoutDeTransport[Interco];
         // flow
         Var = CorrespondanceVarNativesVarOptim->NumeroDeVariableDeLInterconnexion[Interco];
-        if (Var >= 0 && Var < ProblemeAResoudre->NombreDeVariables)
+        if (Var >= 0 && Var < ProblemeAResoudre.NombreDeVariables)
         {
-            ProblemeAResoudre->CoutLineaire[Var] = 0.0;
-            logs.debug() << Var << ". Linear C = " << ProblemeAResoudre->CoutLineaire[Var];
+            ProblemeAResoudre.CoutLineaire[Var] = 0.0;
+            logs.debug() << Var << ". Linear C = " << ProblemeAResoudre.CoutLineaire[Var];
         }
         // direct / indirect flow
         Var = CorrespondanceVarNativesVarOptim
             ->NumeroDeVariableCoutOrigineVersExtremiteDeLInterconnexion[Interco];
-        if (Var >= 0 && Var < ProblemeAResoudre->NombreDeVariables)
+        if (Var >= 0 && Var < ProblemeAResoudre.NombreDeVariables)
         {
             if (TransportCost->IntercoGereeAvecDesCouts == NON_ANTARES)
-                ProblemeAResoudre->CoutLineaire[Var] = 0;
+                ProblemeAResoudre.CoutLineaire[Var] = 0;
             else
-                ProblemeAResoudre->CoutLineaire[Var]
+                ProblemeAResoudre.CoutLineaire[Var]
                     = TransportCost->CoutDeTransportOrigineVersExtremite[hour];
-            logs.debug() << Var << ". Linear C = " << ProblemeAResoudre->CoutLineaire[Var];
+            logs.debug() << Var << ". Linear C = " << ProblemeAResoudre.CoutLineaire[Var];
         }
 
         Var = CorrespondanceVarNativesVarOptim
             ->NumeroDeVariableCoutExtremiteVersOrigineDeLInterconnexion[Interco];
-        if (Var >= 0 && Var < ProblemeAResoudre->NombreDeVariables)
+        if (Var >= 0 && Var < ProblemeAResoudre.NombreDeVariables)
         {
             if (TransportCost->IntercoGereeAvecDesCouts == NON_ANTARES)
-                ProblemeAResoudre->CoutLineaire[Var] = 0;
+                ProblemeAResoudre.CoutLineaire[Var] = 0;
             else
-                ProblemeAResoudre->CoutLineaire[Var]
+                ProblemeAResoudre.CoutLineaire[Var]
                     = TransportCost->CoutDeTransportExtremiteVersOrigine[hour];
-            logs.debug() << Var << ". Linear C = " << ProblemeAResoudre->CoutLineaire[Var];
+            logs.debug() << Var << ". Linear C = " << ProblemeAResoudre.CoutLineaire[Var];
         }
     }
 }
+}
 
-void OPT_InitialiserLesCoutsQuadratiques_CSR(PROBLEME_HEBDO* ProblemeHebdo, int hour)
+ 
+void OPT_InitialiserLesCoutsQuadratiques_CSR(const PROBLEME_HEBDO* ProblemeHebdo,
+                                             PROBLEME_ANTARES_A_RESOUDRE& ProblemeAResoudre,
+                                             int hour)
 {
     logs.debug() << "[CSR] cost";
 
-    PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre = ProblemeHebdo->ProblemeAResoudre;
-    memset((char*)ProblemeAResoudre->CoutLineaire,
+    memset((char*)ProblemeAResoudre.CoutLineaire,
```

Comment:
I know it's not part of the PR, put this cas to `char*` seems suspect to me ?

Reply:
It does